api/watch/funcs.go

```diff
@@ -189,6 +189,13 @@ func checksWatch(params map[string]interface{}) (WatcherFunc, error) {
 		state = "any"
 	}
 
+	var (
+		tags []string
+	)
+	if err := assignValueStringSlice(params, "tag", &tags); err != nil {
```

Comment:
I think I would prefer we take a bexpr filter in here. Users could construct the corresponding filter expression to match tags but then it would be more capable and users could filter on other data as well.

Doing it this way would also alleviate the need for RPC endpoint modifications I think.

Reply:
Thanks for your comment. IIUC, 

>construct the corresponding filter expression... users could filter on other data as well

sounds more like adding a new flag to the `watch` subcommand. So instead of `-tag=some_tag`, you are proposing something like `-filter  'some_tag in ServiceTags'` (`ServiceTags` is the field name in HealthCheck). Is my understanding correct?

Reply:
(continue from my last comment), maybe this is an [example](https://developer.hashicorp.com/consul/api-docs/catalog#filtering-1) of the `filter` flag

Reply:
@mkeeler , replace the `-tag` change in the PR with `-filter`, and it does alleviate the need for RPC endpoint modifications and API client. Thanks.